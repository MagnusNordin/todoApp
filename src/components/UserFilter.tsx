import styled from "styled-components"
import { useAppSelector, useAppDispatch } from "../app/hooks"
import { selectUsers, selectSelectedUser } from "../api/todoSlice"
import { selectUser } from "../api/todoSlice"
import "react-dropdown/style.css"

const ItemWrapper = styled.section`
  display: flex;
  gap: 5px;
  padding-left: 5px;
`
const UserItem = styled.div`
  padding: 2px;
  &.selected {
    border: 2px solid darkgray;
    background-color: lightgray;
  }
`

export const UserFilter = () => {
  const users = useAppSelector(selectUsers)
  const selectedUser = useAppSelector(selectSelectedUser)
  const dispatch = useAppDispatch()

  return (
    <>
      Filtrera på användare
      <ItemWrapper>
        <UserItem
          className={selectedUser === -1 ? "selected" : ""}
          key="all"
          onClick={() => dispatch(selectUser(-1))}
        >
          Alla
        </UserItem>
        {users.map((user) => {
          return (
            <UserItem
              className={selectedUser === user ? "selected" : ""}
              key={user}
              onClick={() => dispatch(selectUser(user))}
            >
              {user}
            </UserItem>
          )
        })}
      </ItemWrapper>
    </>
  )
}
