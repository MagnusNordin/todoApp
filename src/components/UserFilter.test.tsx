import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import { store } from "../app/store"
import { UserFilter } from "./UserFilter"

test("UserFilter", async () => {
  const { asFragment } = render(
    <Provider store={store}>
      <UserFilter />
    </Provider>,
  )

  expect(asFragment).toMatchSnapshot()
})
