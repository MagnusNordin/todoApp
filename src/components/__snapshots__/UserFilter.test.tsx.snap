// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserFilter 1`] = `[Function]`;
